pkg/config/downwardapi.go

```diff
@@ -48,7 +48,12 @@ func CreateDownwardAPIDisks(vmi *v1.VirtualMachineInstance) error {
 			}
 
 			disk := GetDownwardAPIDiskPath(volume.Name)
-			if err := createIsoConfigImage(disk, volume.DownwardAPI.VolumeLabel, filesPath); err != nil {
+			if emptyIso {
+				err = createIsoConfigImage(disk, volume.DownwardAPI.VolumeLabel, filesPath, &vmi.Status.IsoSizes)
+			} else {
+				err = createIsoConfigImage(disk, volume.DownwardAPI.VolumeLabel, filesPath, nil)
+			}
+			if err != nil {
```

Comment:
maybe just something like:
```
var vmiIsoSizes *v1.VirtualMachineInstanceIsoSizes
if emptyIso {
  vmiIsoSizes = &vmi.Status.IsoSizes
}
if err := createIsoConfigImage(disk, volume.DownwardAPI.VolumeLabel, filesPath, vmiIsoSizes); err != nil {
...
```

Reply:
TBH I prefer my version :) But same as above, if you feel strongly about it I'll rewrite those lines.

Reply:
Not strongly.
I think when there is a variable, instead of just `nil` it provides more context to the reader.
Also, you probably don't need this `err` after this function. Up to you :)